renderer/src/GLRenderer.cpp

```diff
@@ -144,7 +151,14 @@ void GLRendererImplementation::start() {
         running_ = true;
     }
 
-    render_loop();
+    try {
+        render_loop();
+    } catch (const std::exception& e) {
+        BOOST_LOG_TRIVIAL(error) << "Exception in main rendering routine " << e.what();
+        stop();
+        if (stop_handler_)
```

Comment:
If there is a `stop` function than, this handler should be invoked inside it I suppose;